core/templates/services/rte-helper-modal.controller.ts

```diff
@@ -245,70 +253,70 @@ export class RteHelperModalComponent {
       }
     }
     this.ngbActiveModal.dismiss(true);
+    this.customizationArgsFormSubscription.unsubscribe();
   }
 
   delete(): void {
     this.ngbActiveModal.dismiss(true);
+    this.customizationArgsFormSubscription.unsubscribe();
   }
 
-  disableSaveButtonForMathRte(): boolean {
-    // This method disables the save button when the Math SVG has not yet
-    // been generated but being processed.
-    if (!this.currentRteIsMathExpressionEditor) {
-      return false;
-    } else {
-      // We know that this is a math rich text component. Hence we can make the
-      // the type more specific.
-      const { value } = this.tmpCustomizationArgs[0] as Extract<
-        CustomizationArgsNameAndValueArray[number],
-        { name: 'math_content' }
-      >;
-      return value.mathExpressionSvgIsBeingProcessed || value.raw_latex === '';
-    }
-  }
-
-  disableSaveButtonForLinkRte(): boolean {
-    // This method disables the save button when the `text` field for the
-    // Link RTE looks like a URL but it does not match the `url`. Otherwise,
-    // creators can make the `url` a malicious website and make the `text`
-    // a safe website.
-    if (!this.currentRteIsLinkEditor) {
-      return false;
-    }
-    // We know that this is a link rich text component. Hence we can make the
-    // the type more specific.
-    const tmpCustomizationArgs = this.tmpCustomizationArgs as Extract<
-      CustomizationArgsNameAndValueArray[number],
-      { name: 'url' | 'text' }
-    >[];
-    let url: string = tmpCustomizationArgs[0].value;
-    let text: string = tmpCustomizationArgs[1].value;
-
-    // First check if the `text` looks like a URL.
-    const suffixes = ['.com', '.org', '.edu', '.gov'];
-    let textLooksLikeUrl = false;
-    for (const suffix of suffixes) {
-      if (text.endsWith(suffix)) {
-        textLooksLikeUrl = true;
+  onCustomizationArgsFormChange(value: number | string | boolean): void {
+    if (this.componentId === this.COMPONENT_ID_MATH) {
+      let rawLatex: string = value[0].raw_latex;
```

Comment:
Same thing, how do we know what value[0] is? Relying on the ordering is not stable, we should try to do this based on the field name. Ditto below.

What is the structure of `value` -- does it give us any info about the field names?

Reply:
Yea this is a problem from the old code, tried to find a solution but the value that is passed is somewhat inconsistent. This relates to the tests created and how the values are updated as well, for example for video args we have 
```
   component.customizationArgsForm.value[1] = 10;
```
as this is the structure of the values that we are subscribing to (video_id, start, end, autoplay)
```
{ "0": "", "1": 0, "2": 0, "3": false }.
```
For math it instead looks like this
```
{
    "0": {
        "raw_latex": "",
        "svg_filename": "",
        "svgFile": null,
        "mathExpressionSvgIsBeingProcessed": false
    }
}
```
tried changing this into also taking the name into consideration as well for mapping customizationArgSpecs
```ts
    this.customizationArgSpecs.forEach((_, index) => {
      formGroupControls[index] = this.fb.control({
        name: this.tmpCustomizationArgs[index].name,
        argValue: this.tmpCustomizationArgs[index].value
      });
    });
```
but it messed up a lot of new and existing logic unfortunately

Reply:
Ah, I see. No worries, thanks a lot for looking into it. Let's just leave it as-is for this PR -- I just realized we do have a little bit of safety since there should be an ordering of fields in the rte config, so this shouldn't break unless someone decides to add a new field in the middle.